python/semantic_kernel/connectors/ai/azure_ai_inference/services/azure_ai_inference_chat_completion.py

```diff
@@ -0,0 +1,270 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+from collections.abc import AsyncGenerator
+from typing import Any
+
+from azure.ai.inference import load_client as load_client_sync
+from azure.ai.inference.aio import ChatCompletionsClient as ChatCompletionsClientAsync
+from azure.ai.inference.models import (
+    AssistantMessage,
+    AsyncStreamingChatCompletions,
+    ChatChoice,
+    ChatCompletions,
+    ChatRequestMessage,
+    ModelInfo,
+    ModelType,
+    StreamingChatChoiceUpdate,
+    SystemMessage,
+    ToolMessage,
+    UserMessage,
+)
+from azure.core.credentials import AzureKeyCredential
+from pydantic import ValidationError
+
+from semantic_kernel.connectors.ai.azure_ai_inference import (
+    AzureAIInferenceChatPromptExecutionSettings,
+    AzureAIInferenceSettings,
+)
+from semantic_kernel.connectors.ai.chat_completion_client_base import (
+    ChatCompletionClientBase,
+)
+from semantic_kernel.contents.chat_history import ChatHistory
+from semantic_kernel.contents.chat_message_content import ChatMessageContent
+from semantic_kernel.contents.streaming_chat_message_content import (
+    StreamingChatMessageContent,
+)
+from semantic_kernel.contents.utils.author_role import AuthorRole
+from semantic_kernel.contents.utils.finish_reason import FinishReason
+from semantic_kernel.exceptions.service_exceptions import ServiceInitializationError
+
+MessageConverter: dict[AuthorRole, Any] = {
+    AuthorRole.SYSTEM: SystemMessage,
+    AuthorRole.USER: UserMessage,
+    AuthorRole.ASSISTANT: AssistantMessage,
+    AuthorRole.TOOL: ToolMessage,
+}
+
+
+class AzureAIInferenceChatCompletion(ChatCompletionClientBase):
+    """Azure AI Inference Chat Completion Service."""
+
+    client: ChatCompletionsClientAsync
+
+    def __init__(
+        self,
+        api_key: str | None = None,
+        endpoint: str | None = None,
+        service_id: str | None = None,
+        env_file_path: str | None = None,
+        env_file_encoding: str | None = None,
```

Comment:
Should we support providing an optional instance of the `ChatCompletionsClientAsync` as a break-glass in case someone needs to override something that we don't break out?

Reply:
Good catch! 
I will add samples in a separate PR that will include the setup of authenticating a client using Entra ID.